src/graph/backend/dnnl/op_executable.hpp

```diff
@@ -2681,7 +2681,10 @@ struct sdpa_executable_t : public op_executable_t {
                 = op->get_input_value(1)->get_logical_tensor().dims[1];
         status_t s = create_sdpa_pd(sdpa_pd_, p_engine.get(), md_q.get(),
                 md_k.get(), md_v.get(), md_dst.get(), md_mask.get(), scale_dt,
-                is_invert_scale_, kv_head_number, is_causal_mask_, attr.get());
+                is_invert_scale_, kv_head_number,
+                is_causal_mask_ ? attn_mask_type::top_left
+                                : attn_mask_type::buffer,
```

Comment:
IIUC, buffer means explicit mask? So do you mean that mask (implicit or explicit) will always exist?

Reply:
When using explicit masks(I can rename it if you think that is a better name), we are testing the memory discriptor passed is null or a pointer to enable the mask operation. I can add another value to the enum value if we want to add checks for this.

Reply:
The name "explicit" doesn't work because its a key word.

Reply:
It would be nice if the mask type name can be more descriptive :)

So, is my following statement correct?
(1) For explicit mask, we need attn_mask_type::buffer && md_mask != null
(2) For implicit (causal) mask, we need attn_mask_type::top_left || attn_mask_type::bottom_right

Reply:
Yes. that is correct.